engine/spverrors/utils.go

```diff
@@ -1,24 +1,20 @@
 package spverrors
 
-import (
-	"fmt"
-
-	"github.com/pkg/errors"
-)
-
 // Wrapf wraps an error with a formatted message
 // If err is nil, Wrapf returns nil.
 func Wrapf(err error, format string, args ...interface{}) error {
-	if len(args) > 0 {
-		return errors.Wrapf(err, format+" caused by", args...)
+	if err == nil {
+		return nil
 	}
-	return errors.Wrap(err, format+" caused by")
+	return NewErrorf(format, args...).Wrap(err)
 }
 
 // Newf creates a new error with a message (which can be formatted)
 func Newf(message string, args ...interface{}) error {
```

Comment:
If this and the above method are calling exported methods and doesn't do anything else, wouldn't it be better to isntead of using `Wrapf` to use `NewErrorf` everywhere? I understand this was made to make less changes in existing code, but what do you think

Reply:
That would be better - and I will change it - but `after` we agree that we want to go for it 😄